cxx/relation.hh

```diff
@@ -21,44 +21,58 @@ class Relation {
  public:
   typedef T ValueType;
 
-  virtual void incorporate(std::mt19937* prng, const T_items& items, ValueType value) = 0;
+  virtual void incorporate(std::mt19937* prng, const T_items& items,
+                           ValueType value) = 0;
 
   virtual void unincorporate(const T_items& items) = 0;
 
-  virtual double logp(const T_items& items, ValueType value, std::mt19937* prng) = 0;
+  virtual double logp(const T_items& items, ValueType value,
+                      std::mt19937* prng) = 0;
 
   virtual double logp_score() const = 0;
 
-  virtual double cluster_or_prior_logp(std::mt19937* prng, const T_items& items, const ValueType& value) const = 0;
+  virtual double cluster_or_prior_logp(std::mt19937* prng, const T_items& items,
+                                       const ValueType& value) const = 0;
 
-  virtual ValueType sample_at_items(std::mt19937* prng, const T_items& items) const = 0;
+  virtual ValueType sample_at_items(std::mt19937* prng,
+                                    const T_items& items) const = 0;
 
-  virtual void incorporate_to_cluster(const T_items& items, const ValueType& value) = 0;
+  virtual void incorporate_sample(std::mt19937* prng, const T_items& items) = 0;
```

Comment:
Please add a description of this new method.

Reply:
Done.